test/metabase/query_processor/middleware/desugar_test.clj

```diff
@@ -1,41 +1,45 @@
-  (is (= {:database 1
-          :type     :query
-          :query    {:source-table 1
-                     :filter       [:and
-                                    [:= [:field 1 nil] "Run Query"]
-                                    [:between
-                                     [:field 2 {:temporal-unit :day}]
-                                     [:relative-datetime -30 :day]
-                                     [:relative-datetime -1 :day]]
-                                    [:!= [:field 3 nil] "(not set)"]
-                                    [:!= [:field 3 nil] "url"]
-                                    [:> [:temporal-extract [:field 4 nil] :year-of-era] 2004]]
-                     :expressions  {"year" [:temporal-extract [:field 4 nil] :year-of-era]}
-                     :aggregation  [[:share [:and
-                                             [:= [:field 1 nil] "Run Query"]
-                                             [:between
-                                              [:field 2 {:temporal-unit :day}]
-                                              [:relative-datetime -30 :day]
-                                              [:relative-datetime -1 :day]]
-                                             [:!= [:field 3 nil] "(not set)"]
-                                             [:!= [:field 3 nil] "url"]]]]}}
-         (desugar/desugar
-          {:database 1
-           :type     :query
-           :query    {:source-table 1
-                      :filter       [:and
-                                     [:= [:field 1 nil] "Run Query"]
-                                     [:time-interval [:field 2 nil] -30 :day]
-                                     [:!= [:field 3 nil] "(not set)" "url"]
-                                     [:> [:get-year [:field 4 nil]] 2004]]
-                      :expressions  {"year" [:get-year [:field 4 nil]]}
-                      :aggregation  [[:share [:and
-                                              [:= [:field 1 nil] "Run Query"]
-                                              [:time-interval [:field 2 nil] -30 :day]
-                                              [:!= [:field 3 nil] "(not set)" "url"]]]]}}))))
+  (binding [qp.timezone/*results-timezone-id-override* "Asia/Ho_Chi_Minh"]
+    (is (= {:database 1
+            :type     :query
+            :query    {:source-table 1
+                       :filter       [:and
+                                      [:= [:field 1 nil] "Run Query"]
+                                      [:between
+                                       [:field 2 {:temporal-unit :day}]
+                                       [:relative-datetime -30 :day]
+                                       [:relative-datetime -1 :day]]
+                                      [:!= [:field 3 nil] "(not set)"]
+                                      [:!= [:field 3 nil] "url"]
+                                      [:> [:temporal-extract [:field 4 nil] :year-of-era] 2004]]
```

Comment:
Add a test for desugaring convert-timezone in the `:filter` clause. I think it will currently fail

Reply:
added here ce7d15314c